app/javascript/utilities/dropdownUtils.js

```diff
@@ -1,3 +1,50 @@
+import { isInViewport } from '@utilities/viewport';
+import { debounceAction } from '@utilities/debounceAction';
+
+/**
+ * Helper function designed to be used on scroll to detect when dropdowns should switch from dropping downwards/upwards.
+ * The action is debounced since scroll events are usually fired several at a time.
+ *
+ * @returns {Function} a debounced function that handles the repositioning of dropdowns
+ * @example
+ *
+ * document.addEventListener('scroll', getDropdownRepositionListener());
+ */
+export const getDropdownRepositionListener = () =>
```

Comment:
Since this is generic enough, we can definitely remove a bunch of code I wrote for the comment subscription dropdown to handle this.

https://github.com/forem/forem/blob/d4ff8516f2ccb2a5200a00ba633a79a426050105/app/javascript/CommentSubscription/CommentSubscription.jsx#L56-L72